### DISCUSSION TITLE
New signup with mariadb

### DISCUSSION THREAD

**Discussion by Zharkan** [nextauthjs/next-auth#1733] [Help] [answered]
**Your question**
How can I prevent new users from signin up ?

**What are you trying to do**
I'm trying to prevent unknown users from signing up. I've tried with callback from the doc but I can't get it to work.

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [x] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by StefanSelfTaught** [comment, marked as answer]
You can try to return false in the signIn callback.

**Reply by Zharkan** [reply]
I'm aware of that but how can I check if a user has an account ?

**Reply by StefanSelfTaught** [reply]
Check for `user.id`

**Reply by Zharkan** [reply]
Should I check in the callback then ?
`  async signIn(userId, account, profile) {
    const isAllowedToSignIn = true
    if (isAllowedToSignIn) {
      return true
    } else {
      return false
    }
  }`

**Reply by StefanSelfTaught** [reply]
Try this:
```js
async signIn(user) {
  if(!user.id) return false;
  return true;  
}

**Reply by Zharkan** [reply]
Yes that's perfect, thank you !
